spec/lib/tasks/trn_generator_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "tasks/trn_generator"
+require "rails_helper"
+
+RSpec.describe TRNGenerator, type: :helper do
+  it "produces a new unassigned value" do
+    expect(::TeacherProfile.pluck(:trn)).not_to include(TRNGenerator.next)
+  end
+
+  it "generates 100_000 unallocated TRNs in under a second" do
+    benchmark = Benchmark.measure do
+      100_000.times do
+        TRNGenerator.next
+      end
+    end
+    expect(benchmark.total).to be < 1
```

Comment:
because of the memoization this is probably why it is so performant. I expect if you wanted actually different TRNs the operation is more expensive for 100k outputs

Reply:
Again. You're missing the `available.pop`. This is performant because it's dealing with an internal array and just treating it like a stack.